src/components/AssetManagementPanel/AssetManagementForm/FormVariants/RejectSurrenderedForm/RejectSurrenderedForm.tsx

```diff
@@ -36,14 +36,15 @@ export const RejectSurrenderedForm: FunctionComponent<RejectSurrenderedFormProps
   restoreTokenState,
 }) => {
   const { showOverlay } = useContext(OverlayContext);
-  const { endorsementChain, pending } = useEndorsementChain(tokenRegistryAddress, tokenId);
+  let lastBeneficiary = "Loading...";
 
-  const lastTransferEvent = endorsementChain
-    ?.filter(({ eventType }) => eventType === "Transfer")
-    .reverse()[0] as TitleEscrowEvent;
-  const lastBeneficiary = lastTransferEvent?.beneficiary;
-  const lastHolderEvent = lastTransferEvent?.holderChangeEvents.reverse()[0];
-  const lastHolder = lastHolderEvent?.holder;
+  const { endorsementChain, pending } = useEndorsementChain(tokenRegistryAddress, tokenId);
+  try {
+    const lastTransferEvent = endorsementChain?.reverse()[0] as TransferEvent;
+    lastBeneficiary = lastTransferEvent?.owner;
+  } catch (e) {
+    console.error(e);
+  }
```

Comment:
do we need to perform the expensive endorsementChain calculation to determine the `lastBeneficiary`?
Is there a cheaper alternative?